canonical-web-and-design/maas-ui
# feat(ui): build used disks and partitions list in machine storage tab

## Done

- Built used disks and partitions list in machine storage tab.
- Refactored a few common elements into utils, and created a helper function to separate machine storage data into disparate bits while only iterating through the disks array once

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the react used disks and partitions list shows the same data as the angular version. You might need a `make sampledata`'d MAAS to see things like volume groups and RAIDs.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2158

## Screenshot
![Screenshot_2020-11-13 honest-corgi ubnt storage focal-maas MAAS](https://user-images.githubusercontent.com/25733845/99037613-c2b48300-25cf-11eb-9719-2ce5f70d78e9.png)
